[Saketh] iP

loose-bus-change opened the pull request:
# Duke
*Your own personal task recording system*

> “Your mind is for having ideas, not holding them.” – David Allen

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ **EXTREMELY FAST** to use

In order to access it:

1. download it from [here](https://github.com/loose-bus-change/ip/releases).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is **FREE!**

Features:

 Managing tasks
 Managing deadlines (coming soon)
 Reminders (coming soon)

This is a sample task list that you can view if you type the `list` command: 
[D][ ] return book    by: 2019-12-02
[E][ ] meeting    at: 2015-10-15
[E][X] project meeting   at: 2015-10-15

Here is the main method of the application:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

loose-bus-change committed:
Finished Level-1

loose-bus-change committed:
Finished Level-2

loose-bus-change committed:
Finished Level-3

loose-bus-change committed:
Finished Level-4

loose-bus-change committed:
Finished A-TextUiTesting

loose-bus-change committed:
Finished Level-5

loose-bus-change committed:
Finished Level-6

loose-bus-change committed:
Finished Level-7

loose-bus-change committed:
Finished Level-8

loose-bus-change committed:
Merge branch 'branch-Level-7'

Merged Level-7

loose-bus-change committed:
Finished Level-8

loose-bus-change committed:
Finished A-More-OOP

loose-bus-change committed:
Finished A-Packages

loose-bus-change committed:
Finished JUnit Testing

loose-bus-change committed:
Finished A-Jar

loose-bus-change committed:
Finish branch-A-JavaDoc

loose-bus-change committed:
Finish A-CodingStandard

loose-bus-change committed:
Finish branch-Level-9

loose-bus-change committed:
Merge branch 'branch-A-JavaDoc'
Committing the javadoc branch

loose-bus-change committed:
Finish A-CodingStandard

loose-bus-change committed:
Merge branch 'branch-Level-9'
Merging the Level 9 branch

gordonlzy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,125 @@
+            s.appendListToFile(t);
+        } else if (input.startsWith("find")) {
+            String keyword = input.substring(5);
+            ArrayList<Task> output = new ArrayList<>();
```
Perhaps the usage of plural form for output will work better?

Smanmos replied:
I think the variable name should be completely renamed. The queried tasks aren't an "output"; they are the result of a query, and are then reformatted into a text result that gets printed.

gordonlzy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,53 @@
+    private Storage storage;
+    private TaskList tasks;
+    private File
+            file;
```
Any reason why you put file on the next line?

gordonlzy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,53 @@
+import duke.*;
```
Would it be better to list all the imported classes?

wilburrito replied:
I concur with @gordonlzy. I think this violates one of the [coding standards.](https://se-education.org/guides/conventions/java/intermediate.html#package-and-import-statements)

Smanmos replied:
I agree as well. Imports should be one at a time.

gordonlzy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,125 @@
+     * @param s The Storage that handles the reading and writing to a text file
+     * @param file The file that gets written to and read from
+     *
+     * @return void
```
Is it necessary to add @return for void?

gordonlzy reviewed (commented):
LGTM, just a few nitpicks

wilburrito commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,125 @@
+     * Method that adds a Deadline task onto TaskList
+     *
+     * @param input The string that contains the user input
+     * @param t The TaskList that contains the tasks to be added
```
perhaps the naming of some of the parameters could be more verbose; however this is just nitpicky, since your javadocs already explain what the parameters mean. Good work! 👍🏻

wilburrito commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,137 @@
+     * @param date The date mentioned in the task initialization.
+     *
+     *
+     * @return void
```
I'm not sure if this is necessary for a `void` method.

wilburrito reviewed (approved):
LGTM, but just a few nits to fix, that's all.

loose-bus-change committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Merge add-gradle-support

JasonC01 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected LocalDate by;
```
In my opinion, deadline seems to be a better name compared to by

JasonC01 reviewed (commented):
LGTM:)!! Great work overall!

Smanmos commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,31 @@
+class NullTaskError extends DukeException {
+    public NullTaskError() {
+        super("OOPS!!! The description of a todo cannot be empty.");
+    }
```
I think it would be better for type to be in the constructor instead of the getMsg method.

Smanmos commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,53 @@
+    private TaskList tasks;
+    private File
+            file;
+    private Parser p;
```
Avoid one letter names; `parser` would be a better name

Smanmos commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,125 @@
+                String[] splitDate = temp[1].split(" ");
+                String date = splitDate[1];
+                String[] breakingDate = date.split("/");
+                String year = breakingDate[2];
+                String month = breakingDate[1];
+                String currentDate = breakingDate[0];
+                int i = Integer.parseInt(currentDate);
+                if (i < 10) {
+                    currentDate = "0" + currentDate;
+                }
+                String finalDateFormat = year + "-" + month + "-" + currentDate;
+                LocalDate date1 = LocalDate.parse(finalDateFormat);
```
Can be simplified using method LocalDate.of

Smanmos commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,38 @@
+
+public class Ui {
+    Scanner scan;
+    private String str;
```
Could be more descriptive; what String is str holding?

Smanmos reviewed (changes requested):
Quite good overall, but many parts could be improved

loose-bus-change committed:
Finish Level-10

loose-bus-change committed:
Fix reading from text file
The reading from file method does not check whether the current task is done or not

This needs to change so that the done nature of the task can be viewed by user

The method of string manipulation has been updated to
include this change

Now every time the user asks for the list, the tasks that are marked done will
be reflected as such in the output

loose-bus-change committed:
Finish A-Assertions

Added assertions to the code

This ensures that there are no unexpected errors in the code
during running

It is being fixed by adding relevant asserts statements
in methods where applicable

loose-bus-change committed:
Finish-A-CodeQuality

loose-bus-change committed:
Merge pull request #1 from loose-bus-change/branch-A-Assertions

Finish A-Assertions

loose-bus-change committed:
Changes

loose-bus-change committed:
Fix conflicts

loose-bus-change committed:
Merge branch 'master' into branch-A-CodeQuality

loose-bus-change committed:
Merge pull request #2 from loose-bus-change/branch-A-CodeQuality

Finish-A-CodeQuality

loose-bus-change committed:
Fix conflict in text file

loose-bus-change committed:
Finish BCD-Extensions

loose-bus-change committed:
Add Ui.png

loose-bus-change committed:
Add User guide

loose-bus-change committed:
Update ReadMe

loose-bus-change committed:
Edit User guide
